### ISSUE TITLE
Fix too low contrast for text background color i.e. yellow highlight against white text

### ISSUE THREAD

**Issue by markusd1984** [darkreader/darkreader#7067]
# Feature Request

## Is this feature request related to a problem?
text highlighted with bright background colors i.e. yellow are very hard to read against white text due to a too low contrast

## Screenshots
![image](https://user-images.githubusercontent.com/24621214/136974116-b1919b9f-51be-4d6a-a8a3-8446aedcffb7.png)

## Additional Context
An additional settings to enable and ideally control text background color contrast would be idea to fix this for either all or individual websites.

I love this plugin and hope this can be solved as I don't want to disable the plugin on sites like Gmail just so I can make text readable. Hoping I didn't miss any option to configure this as I tried the adjustment option in version 5. Only changing from Dynamic to Static works but I loose the overall dark feel that I prefer from the Dynamic mode as Static makes text more brownish rather then white against black and all elements on the website are highlighted.

I noticed there is a custom CSS editor for Static mode. Is there any way i can have the exact same effect as Dynamic mode and just address the text background color issue via css?

the below I think would address this
`[data-darkreader-inline-bgcolor] {color:black}`

**Comment by markusd1984** [comment]
@Gusted yes, text formatting in an email. I can send you tomorrow the html CSS if that helps?

**Comment by stanio** [comment]
Here's possibly more straightforward scenario – non-user-selected style:

```html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Dark highlight test</title>
  <style>
    body {
      color: black;
      background-color: white;
    }
    .hilite {
      background-color: yellow;
      color: black;
    }
  </style>
</head>
<body>

<p>
Bacon ipsum dolor amet boudin dolor ut dolore short ribs pig. Eu
<span class="hilite">capicola meatball jowl short ribs ad</span>
pastrami pariatur porchetta voluptate.</p>

<p>
Bacon ipsum dolor amet boudin dolor ut dolore short ribs pig. Eu
<span style="background-color: yellow; color: black;">capicola
meatball jowl short ribs ad</span> pastrami pariatur porchetta
voluptate.</p>

</body>
</html>
```

Both of the `background-color` and `color` get adjusted by darkreader but the background color is too light for the newly applied foreground color:

![default-style](https://user-images.githubusercontent.com/1247730/137463598-a7e0f857-a894-4117-9ee4-614c25db93e2.png)

![dark-reader](https://user-images.githubusercontent.com/1247730/137463660-3903b98c-9d29-4986-84d5-e6f1711c802e.png)

**Comment by Gusted** [comment]
> Not sure why you focusing on yellow being problematic background color?

Because I'm currently just aware of the low contrast with yellow background colors - I haven't seen any others.



>  Sorry not to go in circles but just to make sure my point is well understood that no matter what background color is used except black, using black text should work because it's always a stronger contrast then white against bright backgrounds.

Yes that indeed should work - on most background-colors the black text is sufficient for a [AA(A) rating](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Perceivable/Color_contrast). Yet that isn't really problem to just implement that - it's a problem that we will then force a static color in our dynamic engine which is what I really want to avoid, as the dynamic engine should be able to create a niece color scheme regardless of original colors(given that the original colors have sufficient contrast otherwise Dark Reader sucks as well)

If we let's say some text color is cyan or light blue on green-ish background-color. That just works and with Dark Reader that works as well and the original colors are preserved throughout the process - by ensuring enough contrast and forcing the text color to be black the original colors are gone and that is something that shouldn't happen with the dynamic engine mode.



> Also re the inline style, is `data-darkreader-inline-bgcolor` targeting anything else besides the "bgcolor" CSS element?

It's only used for CSS property's that target background-colors, so that includes `bgcolor` attribute and the `background-color` property.

**Comment by markusd1984** [comment]
are you saying that we can't detect/differentiate when background-color is used for a text element vs svg? If so than I understand why it's so hard if not even impossible to treat the use cases differently as one thing will always effect another then.

I found this example to see what svg's with background-color looks like and when I inspected it I didn't see the `[data-darkreader-inline-bgcolor]` like it usually is applied for text elements, thus looks to me it's slightly differently treated?

https://codepen.io/noahblon/post/coloring-svgs-in-css-background-images

**Comment by ColonelGerdauf** [comment]
I will comment here to say that I have noticed this issue recently; where the vibrancy of the local background is far too close to the colour of the text. (ie: white-on-yellow). That makes the text downright unreadable.

I would like to know what to do to avoid randomly running into this issue. Would it suffice to catch yellow backgrounds over what was black text and change that to near-brown? Or to simply ignore the conversion of text colour in those specific situations?
